### PR TITLE
[JENKINS-55829] New getter method in AbstractCIBase

### PR DESCRIPTION
**Pull request by nisarg1499**
See [JENKINS-55829](https://issues.jenkins-ci.org/browse/JENKINS-55829).

### Proposed changelog entries

* `Internal` : A new getter method is created in AbstractCIBase and has marked it as `@Restricted(NoExternalUse.class)`.
* Also typecast has been removed as now getter method is in AbstractCIBase, so typecast is of no use.

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests

### Desired reviewers

@oleg-nenashev @daniel-beck

### CODE REVIEW

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Compilation failure needs to be fixed:

```
[ERROR] /home/jenkins/workspace/Core_jenkins_PR-3873/core/src/main/java/hudson/model/AbstractCIBase.java:[96,5] error: cannot find symbol

[ERROR]   symbol:   class Restricted

[ERROR]   location: class AbstractCIBase

```

IIRC there is also a second usage of the class in the AdministrativeMonitor

**Comment by nisarg1499** [comment]
> Compilation failure needs to be fixed:
> 
> ```
> [ERROR] /home/jenkins/workspace/Core_jenkins_PR-3873/core/src/main/java/hudson/model/AbstractCIBase.java:[96,5] error: cannot find symbol
> 
> [ERROR]   symbol:   class Restricted
> 
> [ERROR]   location: class AbstractCIBase
> ```
> IIRC there is also a second usage of the class in the AdministrativeMonitor

Yes, there is a second usage in public void `disable` class. And in that I have done changes.
`Set<String> set = jenkins.getDisabledAdministrativeMonitors();` 
 instead of  `Set<String> set = jenkins.disabledAdministrativeMonitors();`
But same compilation error continues.

File: core/src/main/java/hudson/model/AdministrativeMonitor.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ public void disable(boolean value) throws IOException {
      * he wants to ignore.
      */
     public boolean isEnabled() {
-        return !((AbstractCIBase)Jenkins.get()).disabledAdministrativeMonitors.contains(id);
+        return !(Jenkins.get()).getDisabledAdministrativeMonitors().contains(id);
```
[the remark this answers is not in the document]
Comment (nisarg1499) [reply core/src/main/java/hudson/model/AdministrativeMonitor.java]: I tried your suggested change along with the change mentioned in above comment. But same compilation continues.

**Comment by oleg-nenashev** [comment]
@nisarg1499 are you still interested in finishing it?

**Review by daniel-beck — DISMISSED** [review]
This makes no sense. We do not want to remove the field. This will result in no administrative monitors being considered disabled anymore.

**Comment by nisarg1499** [comment]
> @nisarg1499 are you still interested in finishing it?

Yes I am interested in finishing it. But I am exactly not getting what is the fault in new commit. And confused a little bit too. So can you guide me where the changes went wrong and what should be done?

**Comment by zbynek** [comment]
@nisarg1499 the problem is that `getDisabledAdministrativeMonitors` now returns a new (empty) set each time you call it. So what happens now is 
```java
set=Jenkins.get().getDisabledAdministrativeMonitors(); //gives empty set
set.add(monitor);
Jenkins.get().getDisabledAdministrativeMonitors(); //gives empty set
```
what we want (assuming the set is empty in the beginning) is
```java
set=Jenkins.get().getDisabledAdministrativeMonitors(); //gives empty set
set.add(myMonitor);
Jenkins.get().getDisabledAdministrativeMonitors(); //gives a set containing myMonitor
```
To keep the old functionality you should keep the field `disabledAdministrativeMonitors` and `getDisabledAdministrativeMonitors` should return it.

**Comment by nisarg1499** [comment]
> @nisarg1499 the problem is that `getDisabledAdministrativeMonitors` now returns a new (empty) set each time you call it. So what happens now is
> 
> ```java
> set=Jenkins.get().getDisabledAdministrativeMonitors(); //gives empty set
> set.add(monitor);
> Jenkins.get().getDisabledAdministrativeMonitors(); //gives empty set
> ```
> 
> what we want (assuming the set is empty in the beginning) is
> 
> ```java
> set=Jenkins.get().getDisabledAdministrativeMonitors(); //gives empty set
> set.add(myMonitor);
> Jenkins.get().getDisabledAdministrativeMonitors(); //gives a set containing myMonitor
> ```
> 
> To keep the old functionality you should keep the field `disabledAdministrativeMonitors` and `getDisabledAdministrativeMonitors` should return it.

@zbynek 
Thank you for the explanation. I understood the exact problem and have committed the required changes.

File: core/src/main/java/hudson/model/AbstractCIBase.java [hunk 2]
```diff
@@ -88,12 +90,12 @@ protected void killComputer(Computer c) {
         c.kill();
     }
 
-    /* =================================================================================================================
-    * Package-protected, but accessed API
-    * ============================================================================================================== */
-
-    /*package*/ final CopyOnWriteArraySet<String> disabledAdministrativeMonitors = new CopyOnWriteArraySet<String>();
+	final CopyOnWriteArraySet<String> disabledAdministrativeMonitors = new CopyOnWriteArraySet<String>();
```
Comment (jsoref) [inline core/src/main/java/hudson/model/AbstractCIBase.java]: Not a fan of `<tab>` indentation

Comment (nisarg1499) [reply core/src/main/java/hudson/model/AbstractCIBase.java]: Oh Sorry. I didn't knew that space was there instead of tab !!

**Comment by nisarg1499** [comment]
@oleg-nenashev @daniel-beck 
Please review the changes.

**Review by oleg-nenashev — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
I will remove the label later

**Comment by nisarg1499** [comment]
> I will remove the label later

Okay.

File: core/src/main/java/hudson/model/AbstractCIBase.java [hunk 3]
```diff
@@ -94,6 +96,11 @@ protected void killComputer(Computer c) {
 
     /*package*/ final CopyOnWriteArraySet<String> disabledAdministrativeMonitors = new CopyOnWriteArraySet<String>();
 
+    @Restricted(NoExternalUse.class)
+    public CopyOnWriteArraySet<String> getDisabledAdministrativeMonitors(){
```
Comment (Wadeck) [inline core/src/main/java/hudson/model/AbstractCIBase.java]:
```suggestion
    public @Nonnull CopyOnWriteArraySet<String> getDisabledAdministrativeMonitors(){
```

nice to have ;) (import already done)

**Review by Wadeck — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
Landing without @Wadeck's proposal so that I can squash it properly

**Comment by oleg-nenashev** [comment]
Thanks @nisarg1499 ! Finally got it landed

File: core/src/main/java/hudson/model/AbstractCIBase.java (hunk 3 above)
Comment (daniel-beck) [inline core/src/main/java/hudson/model/AbstractCIBase.java]: Exposes unnecessary implementation detail. Why not `Set`?

**Comment by oleg-nenashev** [comment]
Does not matter much as long as it is restricted

**Pull request merged**